Sources/GRPC/ClientResponseChannelHandler.swift

```diff
@@ -108,10 +111,11 @@ internal class ClientResponseChannelHandler<ResponseMessage: Message>: ChannelIn
   /// are failed with the given status.
   ///
   /// - Parameter status: the status to observe.
-  internal func observeStatus(_ status: GRPCStatus) {
+  internal func observeStatus(_ status: GRPCStatus, trailingMetadata: HTTPHeaders?) {
     if status.code != .ok {
       self.initialMetadataPromise.fail(status)
     }
+    self.trailingMetadataPromise.succeed(trailingMetadata ?? HTTPHeaders())
```

Comment:
Should we instead make the promise's result optional as well?

Reply:
I don't think so: the server will send trailers if it terminates the call (containing the status code at a minimum) so we only have `nil` here if the client terminates the call (by canceling it, timing out, or some other client side error).

Reply:
But what's the advantage of returning an empty `HTTPHeaders` struct to the client user vs. just returning nil?

Reply:
It's easier to deal with an empty collection than a `nil` one.

Reply:
Fair enough :-)